### DISCUSSION TITLE
How to get all functions

### DISCUSSION THREAD

**Discussion by LinuxSuRen** [antonmedv/expr#726] [Q&A] [answered]
As I mentioned in the title. I would like to provide a feature for users to query the functions that they need.

Thanks for the time to read this. I'm looking forward your comment.

**Comment by antonmedv** [comment]
> query the functions that they need

I did not get it. You can use something like Expr Editor which has autocomplete for all expr functions: 

https://expr-lang.org/docs/expr-editor

**Reply by LinuxSuRen** [reply]
My backend language is Golang. So, I'm wondering if there is a way to list all functions in Golang.

**Reply by antonmedv** [reply, marked as answer]
We can do something like this:

```go
import "github.com/expr-lang/expr/builtin"

builtin.Names // list of all builtins
```
